frontends/common/resolveReferences/resolveReferences.cpp

```diff
@@ -287,36 +284,34 @@ bool ResolveReferences::preorder(const IR::Type_Name* type) {
 
 bool ResolveReferences::preorder(const IR::P4Control *c) {
     refMap->usedName(c->name.name);
-    addToContext(c);
     addToContext(c->type->typeParameters);
     addToContext(c->type->applyParams);
     addToContext(c->constructorParams);
+    addToContext(c);  // add the locals
```

Comment:
@jnfoster: This change ensures that parameters enter the scope before the locals.